### ISSUE TITLE
bug: Address bar does not hide when scrolling

### ISSUE THREAD

**Issue by Livijn** [ionic-team/ionic#19081]
# Bug Report

**Ionic version:**
[x] **4.6.2**

**Current behavior:**
When scrolling in a mobile device's browser, the address bar and bottom tab bar remain visible. This is because we are scrolling the `ion-content` and not the `body`. 

**Expected behavior:**
When a user scrolls it should hide the address bar and the bottom tab bar.

**Other information:**
[A common solution to this problem](https://stackoverflow.com/questions/4117377/how-do-i-hide-the-address-bar-on-iphone/7487346#7487346)
[Forum thread](https://forum.ionicframework.com/t/ionic-in-web-hide-browser-address-bar/164970) on the same topic.

**Comment by KevinBassaDevelopment** [comment]
Hi, is there an update of the current state of this problem?

**Comment by IKTANIM** [comment]
Any update on this. I think it is a very important issue. Please fix this issue as soon as possible.

**Comment by ilprima** [comment]
This si very important also for me. Please try to fix this. Maybe something like passing the scroll event from ion-content to the browser to be detected?

**Comment by camicase82** [comment]
Hi guys, any timeframe or estimates on this? personally have used ionic since its first version, and even got along with the fact that I had to rewrite my app code every now and then, but at this point, with clients wanting fully functional PWA, this is becoming a must-have for us and we need to take a decision to keep ionic or move to something else, one last question, is it possible to get its solved through the ionic professional services?

**Comment by ilprima** [comment]
Unfortunately, I think they're avoiding the issue. Now I've been asking for a fix on this for months but still nothing. In mobile the site is really ugly, very old-style, with the scroll that often hangs on iOS.
I know they're working hard.They also released Ionic for React not long ago, but since they sell Ionic as an all-inclusive solution this makes me laugh a bit, because it's absolutely not usable on mobile browser
.

**Comment by chenlitchian** [comment]
HI, anyone can provide screenshot or any link for me to see?

**Comment by ilprima** [comment]
> HI, anyone can provide screenshot or any link for me to see?

Check here: https://ibb.co/GHYr4NL

**Comment by blasco** [comment]
The worst thing is when also using the toolbar, because then we have toobar + browser search bar that doesn't hide...

**Comment by KevinBassaDevelopment** [comment]
I have build a PWA with Angular only, its so clean and everything just works smoothly... Its a pity using Ionic brings these kind of drawbacks.

**Comment by chenlitchian** [comment]
> > HI, anyone can provide screenshot or any link for me to see?
> 
> Check here: https://ibb.co/GHYr4NL

The. site is broken. Can you fix?

**Comment by lluchlluis** [comment]
> I have build a PWA with Angular only, its so clean and everything just works smoothly... Its a pity using Ionic brings these kind of drawbacks.

This is not a good solution if you want to target Google Play and Apple Store.

**Comment by ilprima** [comment]
> > I have build a PWA with Angular only, its so clean and everything just works smoothly... Its a pity using Ionic brings these kind of drawbacks.
> 
> This is not a good solution if you want to target Google Play and Apple Store.

Yes that true. But if it is possible to have a wonderful PWA with Angular, it means we can teach it also on Ionic since it is based on Angular

**Comment by ilprima** [comment]
Please guys, share this thread to all the Ionic users you know and ask them to add a like to the thread so we can let the development team know about that big issue

**Comment by KevinBassaDevelopment** [comment]
> > I have build a PWA with Angular only, its so clean and everything just works smoothly... Its a pity using Ionic brings these kind of drawbacks.
> 
> This is not a good solution if you want to target Google Play and Apple Store.

I use Ionic for Apps that I want to have in the appstore, but this is not always needed.
I mentioned one where I just use Angular because its a great example of how Angular works perfectly for mobile web and mobile wep-apps, and Ionic on top of it reduces this functionality.

**Comment by axebelk** [comment]
Yes, this is really a big issue and I am also badly need this feature. or otherwise I am going to say bye bye to Ionic ;(

**Comment by camicase82** [comment]
> Yes, this is really a big issue and I am also badly need this feature. or otherwise I am going to say bye bye to Ionic ;(

Hi @axebelk what options are you considering? seems like for app store deploy is ionic, react or native?

**Comment by ilprima** [comment]
> Yes, this is really a big issue and I am also badly need this feature. or otherwise I am going to say bye bye to Ionic ;(

I don’t really understand why the authors still avoiding this issue. I asked for a solution 1 year ago!!

**Comment by chenlitchian** [comment]
for website, u can hide the toolbar and tab-bar based on platform
using https://ionicframework.com/docs/angular/platform.
Do anything show hide based on the platform.

**Comment by ilprima** [comment]
> for website, u can hide the toolbar and tab-bar based on platform
> using https://ionicframework.com/docs/angular/platform.
> Do anything show hide based on the platform.

That’s not what requested, toolbar is not the browser status bar and footer that for example in Safari are hidden and shown based on scrolling

**Comment by axebelk** [comment]
> > Yes, this is really a big issue and I am also badly need this feature. or otherwise I am going to say bye bye to Ionic ;(
> 
> Hi @axebelk what options are you considering? seems like for app store deploy is ionic, react or native?

I am consider to use React JS and React native for app development.  :)

**Comment by chenlitchian** [comment]
@liamdebeasi  can we get the update?

**Comment by liamdebeasi** [comment]
I do not have an update at the moment. We will be sure to post here when we have more to share. Thanks!

**Comment by liamdebeasi** [comment]
I posted an update regarding this in another thread, but I wanted to give everyone an update on the status of this issue:

This issue is something we have discussed extensively over the course of both v4 and v5 development. The solution here is not as straightforward as it may seem.

If we had your page's content scrolling on the body, certain visual aspects of your mobile application would be incorrect. For example, since everything is rendered in the body, your `ion-header` and `ion-footer` would not stay fixed to the top/bottom of the screen when you rubber banded (overscroll on iOS). The same goes for `ion-tab-bar`. Since these components are all in your scrollable element, there would be no way for us to fix that (barring some hacky JS solution). I put together a small demo to illustrate some of the visual bugs that appear with this approach: https://codepen.io/liamdebeasi/pen/JjWYRGw.

In addition, the iOS page transition would not likely be possible since it requires that we can isolate the scrollable content from headers and footers. On top of that, other small usability issues pop up such as being able to scroll the content when scrolling on the header and footer. In a native application built with Swift, this should not be possible.

While moving the content onto the body is technically feasible, there are other issues that appear that do not have good solutions. Having the browser vendors expose an API that lets us control this scroll collapsing behavior would allow us to get the best of both worlds, which is why we are in favor of that approach.

We are absolutely interested in resolving this issue if we can come up with a good solution. At the moment, we do not have a good solution that accounts for the other problems I mentioned above.

**Comment by liamdebeasi** [comment]
Hi everyone,

Quick update on this issue:

This issue will be resolved in iOS 15 due to the new mobile Safari interface. We have tested with Ionic apps, and the url bar and tab bar both collapse as you scroll, even if the scroll is not on the body element.

We are going to keep this thread open for now so others can easily find this issue, but we will be closing this out when iOS 15 releases. Thanks for your patience!
